cunumeric/module.py

```diff
@@ -1843,11 +1915,11 @@ def array_split(a, indices, axis=0, equal=False):
                 first_idx = (len_subarr + 1) * res
         split_pts.extend(range(first_idx, end_idx + 1, len_subarr))
     elif (
-        (dtype == np.ndarray and indices.dtype == int)
+        (dtype == np.ndarray and cast(ndarray, indices).dtype == int)
```

Comment:
would need to re-arrange code some for `mypy` to be able to deduce this itself

Reply:
yes, please do. that type for `indices` looks horrendous!

Reply:
https://github.com/nv-legate/cunumeric/pull/413/commits/88b54dd36af1745285d6dd374dcf9ecb1ba0de34